RHDEVDOCS-4781 - Loki (#59285)
openshift/openshift-docs: logging/logging-loki-ocp.adoc

include::_attributes/common-attributes.adoc[]
:context: logging-loki

toc::[]

An essential function of {product-title} is the collection and aggregation of logs from the environments and application pods it runs.

To achieve this, the {logging-title} uses Vector and Loki. Vector is a high-performance observability data pipeline that allows users to configure log forwarders to direct logs to different locations. Loki is a horizontally scalable, highly available, multi-tenant log aggregation system that indexes metadata about logs with labels, similar to Prometheus labels. The log data is compressed and stored in chunks in object stores. Logging is designed to work with the default configuration, which is optimized for small to medium-sized clusters.

In {logging} documentation, *LokiStack* refers to the {logging} supported combination of Loki, and web proxy with {product-title} authentication integration. LokiStack's proxy uses {product-title} authentication to enforce multi-tenancy. *Loki* refers to the object store as either the individual component or an external store. You can query Loki link:https://grafana.com/docs/loki/latest/logql/[using a query language called LogQL].

> include::modules/logging-support-considerations.adoc[leveloffset=+1]

Reviewer: remove in light of new support section? 😉

:_content-type: ASSEMBLY
:context: logging-loki
[id="logging-loki-ocp_{context}"]
= Setting up logging
include::_attributes/common-attributes.adoc[]

toc::[]

> An essential function of {product-title} is the collection and aggregation of logs from the environments and application pods it runs.

Author: Product title is OCP though, not logging. So *the* over *an* would be a stretch :)

> 

Reviewer 2: @periklis  stored in chunks in object stores or locally on the filesystem.  loki-operator  doesn't support data on filesystem.  Shall we update it?

To achieve this, the {logging-title} uses Vector and Loki. Vector is a high-performance observability data pipeline that allows users to configure log forwarders to direct logs to different locations. Loki is a horizontally scalable, highly available, multi-tenant log aggregation system that indexes metadata about logs with labels, similar to Prometheus labels. The log data is compressed and stored in chunks in object stores or locally on the filesystem. Logging is designed to work with the default configuration, which is optimized for small to medium-sized clusters.

> In {logging} documentation, *LokiStack* refers to the {logging} supported combination of Loki, and web proxy with {product-title} authentication integration. LokiStack's proxy uses {product-title} authentication to enforce multi-tenancy. *Loki* refers to the log store as either the individual component or an external store. You can query Loki link:https://grafana.com/docs/loki/latest/[using JSON paths or regular expressions].

Reviewer 3: > You can query Loki link:https://grafana.com/docs/loki/latest/[using JSON paths or regular expressions].

To me it sounds like this sentence should instead refer to the "LogQL" query language used in Loki. Destructuring JSON and matchers using regular-expressions are a part of it, but you can't directly throw JSONPath or a regex at the query engine:

> You can query Loki link:https://grafana.com/docs/loki/latest/logql/[using a query language called LogQL].

include::modules/logging-support-considerations.adoc[leveloffset=+1]

[id="loki-deployment-sizing_{context}"]
== Deployment Sizing

include::snippets/logging-loki-size-snip.adoc[]

include::modules/logging-loki-gui-install.adoc[leveloffset=+]

include::modules/logging-loki-cli-install.adoc[leveloffset=+1]

=== Troubleshooting installation issues

include::snippets/logging-install-trouble-snip.adoc[]

== Creating a secret for your log store

include::modules/logging-loki-storage.adoc[leveloffset=+1,lines=9..278]

== Creating the `LokiStack` CR

> 1. Click on `Operators` and then `Installed Operators`.
> 

Reviewer 4: Generally, use **bold** for UI elements and monospace font for commands, paths, object names, etc.
Maybe:
```suggestion
1. Click *Operators* and then *Installed Operators*.
```
If so, multiple instances in this PR.

  a. Select the `Loki Operator`.

  b. On the first page under `Provided APIs` and `LokiStack` select `Create instance`.

  c. Switch to `YAML view` option.

  d. Replace the YAML present using the template below:

include::snippets/logging-lokistack-cr-snip.adoc[]

  e. Then click `Create`.

== Creating the `ClusterLogging` CR instance

1. Click on `Operators` and then `Installed Operators`.

  a. Select the `Red Hat OpenShift Logging` Operator.

  b. On the first page under `Provided APIs` and `Cluster Logging`, select `Create instance`.

  c. Replace the YAML present using the template below:

include::snippets/logging-clusterlogging-cr-snip.adoc[]

> This will create an instance of `ClusterLogging` within the namespace `openshift-logging`.

Reviewer 4: IBM Style: Avoid future tense. Replace `will <verb>` with the present tense of the verb. About 25 instances of "will" in the PR.

d. Click `Create`.

=== Verify the Logging install

> Now that Logging has been created, verify the install.

Reviewer 4: Maybe use "installation" instead of "install."
```suggestion
Now that Logging has been created, verify the installation.
```
Multiple instances in this PR.

> 1. Switch to the `Workloads` → `Pods` page.
> 
> 2. Select the `openshift-logging` project.

Reviewer 4: ```suggestion
. Switch to the `Workloads` → `Pods` page.

. Select the `openshift-logging` project.
```
Many instances in this PR.

> You should see pods for `cluster logging` (the operator itself), the `collectors`, `logging-view-plugin`, and a variety of `lokistack` pods.

Reviewer 4: The localization team also complained about the ambiguity of "should" and asked us to avoid it.

3. Verify that you see the following pods: *cluster logging*, *collectors*, *logging-view-plugin*, and various *lokistack* pods.

4. Shortly after installation, a box in the top right corner displays "Web console update is available" and prompts you to refresh your browser. Refresh the page and verify that you see the logs.

== Observing the logs

1. Select *Observe* -> *Logs* on the left menu.

2. Select *Infrastructure*.

> 3. Verify that you see the *Infrastructure* logs grouped in *application*, *infrastructure*, and *audits* sections.

Reviewer 3: This does not make sense to me. "Infrastructure" is one of our "log types" ("tenants"?). It does not include "application" and "audit" but is on the same "level" as those.

Reviewer 5: @libander I think this is not updated yet.

[NOTE]
====

> `audit` logs are not captured by default.

Reviewer 3: I think we always use "log collection":

```suggestion
`audit` logs are not collected by default.
```

You can filter by `Content`, `Namespaces`, `Pods`, and `Containers`.
This can be useful to narrow down searches when looking for something more specific.

You can further specify the logs you are looking for by using the other

> drop down menu for `Severity`. This menu breaks the logs down into `critical`, `error`, `warning`, `debug`, `info`, `trace`, and `unknown` logging categories.

Reviewer 4: IBM Style: drop-down adjective

> The final section is the *histogram*. This gives you a more visual format for viewing log data.

Reviewer 3: Do we need to describe what the histogram does or is this obvious for the user?

```suggestion
The final section is the *histogram*. This shows how many log entries match the given query over time and can give you an indication when events happened more frequently.
```

=== Accessing audit logs

> To have access to `audit logs`, log forwarding must be setup.

Reviewer 4: ```suggestion
To have access to `audit logs`, log forwarding must be set up.
```

1. In the {product-title} web console, click *Operators* -> *OperatorHub*.

2. Select *Red Hat OpenShift Logging*.

3. Select *Provided APIs* then *Cluster Log Forwarder* and click *Create instance*.

4. Replace the YAML displayed with the following template:

[source,yaml]
----

> - apiVersion: logging.openshift.io/v1

Reviewer 3: ```suggestion
apiVersion: logging.openshift.io/v1
```
